roles/deploy/tasks/main.yml

```diff
@@ -33,9 +33,18 @@
 - name: deploy to production directories with capistrano
   shell: cap {{ deploy_target }} deploy chdir=/home/{{ ansible_ssh_user }}/{{ project_name }}
 
-- name: restart tomcat and apache
+- name: restart tomcat-solr and apache
   become: true
+  when: solr_main_version == 4
   service: name={{ item }} state=restarted
   with_items: 
     - tomcat7
     - apache2
```

Comment:
Are the solr and tomcat restarts in any way linked?  I might be clearer to refactor this to 
 - name: restart apache
    become: true
    service: name=apache2 state=restarted

 -  name: restart solr 4.x
    become: true
    service: name=tomcat7 state=restarted
    when: solr_main_version == 4

 -  name: restart solr 5.x
    become: true
    service: name=solr state=restarted
    when: solr_main_version == 5